flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/main/java/org/apache/flink/tests/util/kafka/LocalStandaloneKafkaResource.java

```diff
@@ -251,7 +251,11 @@ public void sendMessages(String topic, String... messages) throws IOException {
 			"--broker-list",
 			KAFKA_ADDRESS,
 			"--topic",
-			topic)) {
+			topic,
+			"--property",
+			"parse.key=true",
+			"--property",
+			"key.separator=\t")) {
```

Comment:
baking this directly into this method smells a bit. It is not obvious to the user that this format is expected. What happens if no key is provided, like in the SQLClientKafkaITCase?

Reply:
Nothing happens, because sending without key still works. I could add a second method for sending data with key.